src/Microsoft.Identity.Client/TokenCache.cs

```diff
@@ -101,7 +101,24 @@ internal void OnBeforeWrite(TokenCacheNotificationArgs args)
             BeforeWrite?.Invoke(args);
         }
 
-        internal AccessTokenCacheItem SaveAccessAndRefreshToken(AuthenticationRequestParameters requestParams,
+        internal AccessTokenCacheItem SaveAccessAndRefreshToken(AuthenticationRequestParameters requestParams, TokenResponse response)
+        {
+            var atCacheEvent = new CacheEvent(CacheEvent.TokenCacheWrite) { TokenType = CacheEvent.TokenTypes.AT };
+            Telemetry.GetInstance().StartEvent(requestParams.RequestContext.TelemetryRequestId, atCacheEvent);
+            var rtCacheEvent = new CacheEvent(CacheEvent.TokenCacheWrite) { TokenType = CacheEvent.TokenTypes.RT };
+            Telemetry.GetInstance().StartEvent(requestParams.RequestContext.TelemetryRequestId, rtCacheEvent);
```

Comment:
This should not work, we cannot have 2 events of the same type and same request id going on at the same time.